### PR TITLE
Update Form and TextInput

### PR DESCRIPTION
**Pull request by jonnystening**
## Updates
- Use new tokens
- Change boolean isPropname syntax,
- Change the way FormContext is created and consumed

Note: Form sub-components (e.g. TextInput) can be used without a Form wrapper so we may not always have FormContext. Removed the throw error from @simonswiss's Context API example in this case.

### CODE REVIEW

File: components/form/src/ErrorMessage.js [hunk 1]
```diff
@@ -43,17 +35,17 @@ export const ErrorMessage = ({ message, icon, tag: Tag, ...props }) => {
 	}
 
 	const common = {
-		fontSize: errorMessage.fontSize,
-		margin: errorMessage.margin,
-		color: errorMessage.color,
+		fontSize: '0.875rem',
+		margin: '0 0 0.75rem',
+		color: COLORS.danger,
 		...(isMessages && { listStyle: 'none', paddingLeft: 0 }),
```
Comment (jaortiz) [inline components/form/src/ErrorMessage.js]: can do in one line
```
...(Array.isArray(message) && { listStyle: 'none', paddingLeft: 0 }), 
```

File: components/form/src/Hint.js [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
 	};
 
-	return <Tag css={common} {...props} />;
+	return (
+		<Tag
+			css={{
+				color: COLORS.muted,
+				fontSize: '0.875rem',
+				marginTop: spacingMap[spacing].marginTop,
+				marginBottom: spacingMap[spacing].marinBottom,
```
Comment (jaortiz) [inline components/form/src/Hint.js]:
```
...spacingMap[spacing]
```

File: components/form/src/ErrorMessage.js (hunk 1 above)
Comment (jonnystening) [reply components/form/src/ErrorMessage.js]: It's used above though, saves calling isArray twice... thought the variable and comment also made it clearer

File: components/form/src/Hint.js (hunk 2 above)
Comment (jonnystening) [reply components/form/src/Hint.js]: whoops also what's a `marinBottom` haha

File: components/form/src/ErrorMessage.js (hunk 1 above)
Comment (jaortiz) [reply components/form/src/ErrorMessage.js]: Didnt see it being used above, github had it collapsed. Just inline the common styles

**Pull request merged**